backend/src/helpers/auth.js

```diff
@@ -1,7 +1,46 @@
-const authPreHandler = async (isAuth) => {
-    if ( isAuth ) {
-        return true;
-    }
-}
+const jwksClient = require('jwks-client');
+const jwt = require('jsonwebtoken');
 
-module.exports = authPreHandler;
\ No newline at end of file
+/**
+ * Validate the bearer token passed with an API request.
+ * 
+ * @param {*} req 
+ * @returns {}
+ */
+const validateToken = (jwksUri, req) => {
+    return new Promise((resolve, reject) => {
+        const token = req.headers.authorization;
+        if (token && token.indexOf('Bearer ') == 0) {
+            const tokenString = token.split(' ')[1];
+            console.log(tokenString)
+            const kid = jwt.decode(tokenString, { complete: true }).header.kid;
+            const client = jwksClient({
+                strictSsl: true,
+                jwksUri: jwksUri
+            });
+            
+            // Get the public key from the keycloak server which is used to verify the JWT.
+            client.getSigningKey(kid, (err, key) => {
+                if (err) {
+                    reject(new Error(`Signing key error. ${err}`));
```

Comment:
for mvp this is fine, no worries here. maybe we should think about collecting these error messages so that if one day a user encounters them, we don't get bug tickets

Reply:
Totally agree, we should actually be using the logging features of fastify...

Reply:
Agreed and when any one of these errors is thrown, it shows up in the node logs already. To be clear, I don't think a user should ever have the opportunity to encounter authentication-related logs. Theoretically, we don't want to be revealing exactly what went wrong with a bad-faith authentication attempt. The API currently just returns a 401 with message "Not Authorized" and I think that's the way it should stay